lib/utils.pm

```diff
@@ -2300,7 +2300,7 @@ sub permit_root_ssh_in_sol {
     my $sshd_config_file = shift;
 
     $sshd_config_file //= "/etc/ssh/sshd_config";
-    enter_cmd("[ `grep \"^PermitRootLogin *yes\" $sshd_config_file | wc -l` -gt 0 ] || (echo 'PermitRootLogin yes' >>$sshd_config_file; systemctl restart sshd)");
+    enter_cmd("[ `grep \"^PermitRootLogin *yes\" $sshd_config_file | wc -l` -gt 0 ] || (echo 'PermitRootLogin yes' >>$sshd_config_file; systemctl restart sshd)", wait_still_screen => 5);
```

Comment:
You are using `sleep()` or are simulating a sleep-like functionality. This is introducing the risk of either unnecessarily wasting time or just shifting race conditions and making them even harder to investigate. You *can* use sleep but there are only very limited cases where is the best solution (e.g. yield threads in low-level system code). Here you have the classical problem in software design of [Synchronization](https://en.wikipedia.org/wiki/Synchronization_(computer_science)) . sleep should not be used in place of proper synchronization methods. https://blogs.msmvps.com/peterritchie/2007/04/26/thread-sleep-is-a-sign-of-a-poorly-designed-program/ is one of multiple references that explains this: "The original problem is likely a timing/synchronization issue, ignoring it by hiding it with Thread.Sleep is only going to delay the problem and make it occur in random, hard to reproduce ways." And if the problem does not occur at least time is wasted. https://testing.googleblog.com/2021/03/test-flakiness-one-of-main-challenges.html states that as well: "Do NOT add arbitrary delays as these can become flaky again over time and slow down the test unnecessarily."

We are waiting because some condition changes some time ... keyword(s) is/are *some time*. It is important to know the exact condition that the code should wait for and check for that instead of delaying program execution for an arbitrary amount of time. For openQA tests in general just try to do what you as a human user would also do. You would not look on your watch waiting for exactly 100s before you look at the screen again, right? :)

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details

Reply:
Thank you for the detailed explaination. I understand 'sleep' should be avoided in our tests as we could. I explain why it had to be used here:

- The command is running in SOL console to start sshd service. It is necessary because ssh root console is not available until sshd service is restarted.
- SOL console(ipmi Serial Over Lan console) has very limited support in openQA, fg. script_output() and assert_script_run() are not supported well. It means any checks for some conditions are not easy to be implemented and even with some unexpected results.
- SOL console is not stable. We avoid run command on it as we could as possible. An extra check would make the test be at risk for failing.
- I have monitored in a few test cases, simply waiting for 5s will make sure sshd service is brought up. 

So I think 'sleep' in this PR is a smart move.